android/crypto/src/main/java/com/coinbase/wallet/crypto/ciphers/KeyStores.kt

```diff
@@ -0,0 +1,117 @@
+package com.coinbase.wallet.crypto.ciphers
+
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import java.io.IOException
+import java.security.KeyStore
+import java.security.KeyStoreException
+import java.security.NoSuchAlgorithmException
+import java.security.UnrecoverableEntryException
+import java.security.cert.CertificateException
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import kotlin.concurrent.withLock
+
+object KeyStores {
+    /**
+     * Get a key spec for gcm block mode without requiring user authentication
+     *
+     * @param alias the alias of the key
+     * @return a key spec for gcm block mode without user authentication
+     */
+    fun getGCMKeySpec(alias: String): KeyGenParameterSpec = getKeyGenParameterSpecBuilder(alias).build()
+
+    /**
+     * Get a key spec for gcm block mode and require user authentication.
+     *
+     * @param alias the alias of the key
+     * @return a key spec for gcm block mode with user authentication
+     */
+    fun getGCMWithUserAuthenticationKeySpec(alias: String): KeyGenParameterSpec = getKeyGenParameterSpecBuilder(alias)
```

Comment:
nit: merge with `buildGCMKeySpec` above and just pass a parameter i.e.

```kotlin
fun buildGCMKeySpec(alias: String, isAuthenticated: Boolean = false)
```

Reply:
3bc1ba8d443523dafcce732a1d79390218c1bb1d